src/main/java/seedu/dietmanager/logic/commands/profile/CheckWeightRecordCommand.java

```diff
@@ -66,12 +68,31 @@ public Result getResult(Profile profile) {
         initialWeight = weightRecord.get(0);
         currentWeight = weightRecord.get(weightRecord.size() - 1);
         weightDifference = initialWeight - currentWeight;
+        weightToGoal = abs(weightGoal - currentWeight);
         if (weightDifference > 0) {
-            this.resultString = String.format(MessageBank.WEIGHT_LOSS_MESSAGE, weightDifference);
+            if (currentWeight == weightGoal) {
+                this.resultString = String.format(MessageBank.WEIGHT_LOSS_MESSAGE
+                        + MessageBank.WEIGHT_GOAL_ACHIEVED_MESSAGE, weightDifference, weightToGoal);
+            } else {
+                this.resultString = String.format(MessageBank.WEIGHT_LOSS_MESSAGE
+                        + MessageBank.WEIGHT_GOAL_NOT_ACHIEVED_MESSAGE, weightDifference, weightToGoal);
+            }
         } else if (weightDifference == 0) {
-            this.resultString = MessageBank.WEIGHT_NO_CHANGE_MESSAGE;
+            if (currentWeight == weightGoal) {
+                this.resultString = String.format(MessageBank.WEIGHT_NO_CHANGE_MESSAGE
+                        + MessageBank.WEIGHT_GOAL_ACHIEVED_MESSAGE, weightDifference, weightToGoal);
+            } else {
+                this.resultString = String.format(MessageBank.WEIGHT_NO_CHANGE_MESSAGE
+                        + MessageBank.WEIGHT_GOAL_NOT_ACHIEVED_MESSAGE, weightDifference, weightToGoal);
+            }
         } else {
-            this.resultString = String.format(MessageBank.WEIGHT_GAIN_MESSAGE, abs(weightDifference));
+            if (currentWeight == weightGoal) {
+                this.resultString = String.format(MessageBank.WEIGHT_GAIN_MESSAGE
+                        + MessageBank.WEIGHT_GOAL_ACHIEVED_MESSAGE, abs(weightDifference), weightToGoal);
+            } else {
+                this.resultString = String.format(MessageBank.WEIGHT_GAIN_MESSAGE
+                        + MessageBank.WEIGHT_GOAL_NOT_ACHIEVED_MESSAGE, abs(weightDifference), weightToGoal);
+            }
```

Comment:
This portion is a bit long, maybe afterward can try to extract the method. Or split it up into different cases